### PR TITLE
Change type of layer Metadata from dict to Mapping

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

same as https://github.com/napari/npe2/pull/357

# Description

Implement [napari.zulipchat.com#narrow/stream/212875-general/topic/DeprecatingDict.20not.20accepted.20in.20LayerDataTuple/near/468436819](https://napari.zulipchat.com/#narrow/stream/212875-general/topic/DeprecatingDict.20not.20accepted.20in.20LayerDataTuple/near/468436819)

### CODE REVIEW

File: napari/plugins/io.py [hunk 1]
```diff
@@ -77,8 +77,9 @@ def read_data_with_plugins(
 
     res = _npe2.read(paths, plugin, stack=stack)
     if res is not None:
-        _ld, hookimpl = res
-        return [] if _is_null_layer_sentinel(_ld) else _ld, hookimpl
+        ld_: Sequence[LayerData]
+        ld_, hookimpl = res
+        return [] if _is_null_layer_sentinel(ld_) else list(ld_), hookimpl
```
Comment (Czaki) [inline napari/plugins/io.py]: Workaround for https://github.com/python/mypy/issues/17752

File: napari/components/viewer_model.py [hunk 2]
```diff
@@ -1587,12 +1587,12 @@ def _unify_data_and_user_kwargs(
 ) -> FullLayerData:
     """Merge data returned from plugins with options specified by user.
 
-    If ``data == (_data, _meta, _type)``.  Then:
+    If ``data == (data_, meta_, type_)``.  Then:
```
Comment (DragaDoncila) [inline napari/components/viewer_model.py:1590]: I assume putting the underscore at the end is to appease some formatting rule?

**Review by DragaDoncila — APPROVED** [review]
Looks fine to me!

File: napari/components/viewer_model.py (hunk 2 above)
Comment (Czaki) [reply napari/components/viewer_model.py:1590]: The `_name` is pattern for unused named variable.

**Pull request merged**